2200073/1949_등산로_조성.py

```diff
@@ -0,0 +1,180 @@
+# 1949 등산로 조성
+
+# 여러 방법으로 풀어봤는데 계속 틀렸다
+# 어떤 케이스는 맞고 어떤 케이스는 틀리고..
+# 51개 케이스 중 41개 정답..
+######## 푸는중 ###########
+
+import sys
+sys.stdin = open('input_1949.txt', 'r')
+
+def dfs(x,y):
+    global ans
+    stack = [(x, y)]
+    visited[x][y] = 1
+    while stack:
+        x, y = stack.pop()
+        ans = max(ans, visited[x][y])
+        for idx in range(4):
+            nx = x + dx[idx]
+            ny = y + dy[idx]
+            if 0 <= nx < n and 0 <= ny < n:
+                if not visited[nx][ny] and graph[nx][ny] < graph[x][y]:
+                    visited[nx][ny] = visited[x][y] + 1
+                    stack.append((nx, ny))
+        visited[x][y] = 0
+
+T = int(input())
+
+# 상하좌우
+dx = [-1, 1, 0, 0]
+dy = [0, 0, -1, 1]
+
+for t in range(T):
+    ans = 0
+    n, k = map(int, input().split())
+    graph = [list(map(int, input().split())) for _ in range(n)]
+    for a in range(1, k+1):
+        for i in range(n):
+            for j in range(n):
+                graph[i][j] -= a
+                max_ = max(map(max,graph))
+                for x in range(n):
+                    for y in range(n):
+                        if graph[x][y] == max_:
+                            visited = [[0] * n for _ in range(n)]
+                            dfs(x, y)
+                graph[i][j] += a
```

Comment:
```suggestion
    graph = [list(map(int, input().split())) for _ in range(n)]
    max_ = max(map(max,graph))
    for a in range(1, k+1):
        for i in range(n):
            for j in range(n):
                graph[i][j] -= a
                for x in range(n):
                    for y in range(n):
                        if graph[x][y] == max_:
                            visited = [[0] * n for _ in range(n)]
                            dfs(x, y)
                graph[i][j] += a
```
문제 서술이 명확하지 않아 문제의 의도와 다르게 이해하신 부분이 있는 것 같습니다 😢
문제의 의도는, 탐색을 시작하는 봉우리는 초기 상태의 가장 높은 위치에서 시작하는 것이었습니다. 즉, 어느 지점을 깎아 최고 지점 봉우리들의 집합이 바뀌더라도, 출발 가능한 지점은 바뀌지 않도록 수정하는 것이 문제의 의도였습니다.